src/h5web/visualizations/line/DataCurve.tsx

```diff
@@ -19,45 +23,82 @@ function DataCurve(props: Props): ReactElement {
 
-  const dataGeometry = useMemo(() => {
-    const points = ordinates.map((val, index) => {
-      const ordinate = ordinateScale(val);
-      return new Vector3(
-        abscissaScale(abscissas[index]),
-        // This is to avoid a three.js warning when ordinateScale(val) is Infinity
-        Number.isFinite(ordinate) ? ordinate : 0,
-        // Move NaN/Infinity out of the camera FOV (negative val for logScale).
-        // This allows to have only curve segments for the positive values
-        Number.isFinite(ordinate) ? 0 : camera.far
-      );
-    });
+  const points = useMemo(
+    () =>
+      ordinates.map<{ data: Vector3; minErr?: Vector3; maxErr?: Vector3 }>(
+        (val, index) => {
+          const x = abscissaScale(abscissas[index]);
+          const y = ordinateScale(val);
 
-    const geometry = new BufferGeometry();
-    geometry.setFromPoints(points);
-    return geometry;
-  }, [abscissaScale, abscissas, camera, ordinateScale, ordinates]);
+          // Move NaN/Infinity out of the camera FOV (negative val for logScale).
+          // This allows to have only curve segments for the positive values
+          const finiteData = Number.isFinite(x) && Number.isFinite(y);
+          const data = finiteData
+            ? // Set x,y to 0 to avoid a three.js warning when one is Infinity
+              new Vector3(x, y, 0)
+            : // Move NaN/Infinity out of the camera FOV (negative val for logScale).
+              // This allows to have only curve segments for the positive values
+              new Vector3(0, 0, camera.far);
+
+          if (!errors || !errors[index] || !finiteData) {
+            return { data };
+          }
+          const yErrBottom = ordinateScale(val - errors[index]);
+          const yErrTop = ordinateScale(val + errors[index]);
+
+          return {
+            data,
+            minErr: Number.isFinite(yErrBottom)
+              ? new Vector3(x, yErrBottom, 0)
+              : undefined,
+            maxErr: Number.isFinite(yErrTop)
+              ? new Vector3(x, yErrTop, 0)
+              : undefined,
+          };
+        }
+      ),
+    [abscissaScale, abscissas, camera.far, errors, ordinateScale, ordinates]
+  );
+
+  const ref = useUpdate(
+    (geometry: BufferGeometry) =>
+      geometry.setFromPoints(points.map((points) => points.data)),
+    [points]
+  );
 
   const showLine = curveType !== CurveType.GlyphsOnly;
   const showGlyphs = curveType !== CurveType.LineOnly;
 
   return (
     <Suspense fallback={<></>}>
-      <DataGlyphs
-        geometry={dataGeometry}
-        color={color}
-        visible={showGlyphs}
-        size={6}
-      />
-      <Line geometry={dataGeometry} visible={showLine}>
+      <Line visible={showLine}>
         <lineBasicMaterial attach="material" color={color} linewidth={2} />
+        <bufferGeometry attach="geometry" ref={ref} />
       </Line>
+      <points visible={showGlyphs} geometry={ref.current}>
```

Comment:
As the `geometry` is set differently here and in `ErrorBars`, I moved the `<points /> out of `DataGlyphs` to keep only the sprite material (now `GlyphMaterial`)